src/odemis/acq/stream/_sync.py

```diff
@@ -1235,6 +1235,13 @@ def _adjustHardwareSettings(self):
             # CCD to be sure it is not slowing thing down.
             self._emitter.dwellTime.value = self._emitter.dwellTime.clip(exp + readout)
 
+        if model.hasVA(self._emitter, "external") and self._emitter.external.value is None:
+            # When the e-beam is set to automatic external mode, it would switch on/off for every
+            # block of acquisition. This is not efficient, and can disrupt the e-beam. So we force
+            # "external" while the acquisition is running.
+            self._orig_hw_values[self._emitter.external] = self._emitter.external.value
```

Comment:
I saw that in some microscope software, the user could explicitly set the control mode to external. Does this change make that unnecessary now?

Reply:
No. It's the same functionality. On some hardware Odemis is able to control it. On these hardware, on some SPARC acquisition modes, this external mode would switch on/off rapidly, corresponding to each pixel. That fixes it.